### ISSUE TITLE
Folder Organization Suggestions

### ISSUE THREAD

**Issue by amcdnl** [aws-amplify/amplify-cli#2289]
## Request
Improve the organization of source code and amplify generated files.

## Issue
Currently when you generate a project with the Amplify CLI, it can be very difficult to follow the code because it mixes source code, dependencies, configuration (often generated) and built files. Additionally, when you have deeply nested projects managing the node_modules and nested scripts are difficult.

## Solutions
[Yarn Packages](https://yarnpkg.com/lang/en/docs/workspaces/) make organizing this type of things really easy. I have a project that I built with the AWS CDK and Yarn Workspaces and the end result was very clean. The project is broke down into a few different packages:

- `api` - lambda + api gateway
- `aws` - CDK generation code
- `web` - create react app
- `notification` - bot notification lambda
- `aggregator` - data aggregation lambda

![2019-09-14_09-13-00](https://user-images.githubusercontent.com/227909/64908562-f811c800-d6cf-11e9-8107-0c30fc6bfba9.png)

The organization of the generated code is nicely contained in the `cdk.out` folder and all my source code just is consumed by the `aws` project here. Now if you compare this to what one of my AWS Amplify projects looks like, its very difficult to understand...

![2019-09-14_09-13-00](https://user-images.githubusercontent.com/227909/64908620-afa6da00-d6d0-11e9-9811-e5ceaa2e9372.png)

The Firebase examples aren't as 'clean' as the workspaces but they are still easier to follow. Here is one example: https://github.com/firebase/friendlychat-web

**Comment by visusnet** [comment]
I would also love to use yarn workspaces but unfortunately, it is not very easy to implement. The problem is that yarn hoists dependencies and symlinks them. I didn‘t find an acceptable solution that creates Lambda bundles. Internally, the npm package archiver is used to create zip files but this library does not support symlinks. They are simply ignored. I didn‘t find an alternative to archiver that could be used as a replacement (which does not mean there is none). As long as this is not possible, yarn workspaces are not really working. There are some people who wrote that they got it working using hacks but I wasn‘t able to reproduce their results (see https://github.com/aws-amplify/amplify-cli/issues/1696).

However, even without yarn workspaces, it would be possible to create a better folder structure. One could even implement a yarn workspaces look-alike folder structure (/packages/<resourceName> instead of /amplify/backend/<category>/<resourceName>.

To summarize: yarn workspaces won‘t work at the moment and a better folder structure would be great, even without workspaces.

**Comment by amcdnl** [comment]
For my CDK project, I ended up doing this:

```
"workspaces": {
    "nohoist": [
      "aws-sdk",
      "aws-sdk/**"
    ]
  }
```

I'm sure there is a better way but it worked so didn't put too much more time into it.

**Comment by attilah** [comment]
@amcdnl thanks for the suggestion, we'll look at it. This is quite a big change and would require a great amount of refactoring all over the packages.

**Comment by hogarthww-labs** [comment]
I created this issue https://github.com/aws-amplify/amplify-cli/issues/6247 to address the folder structure problem
